File: app/src/main/java/com/bq/zowi/models/commands/StaticCommand.java
package com.bq.zowi.models.commands;

import com.bq.zowi.models.commands.Command;

/* JADX INFO: loaded from: classes.dex */
public class StaticCommand extends MovementCommand {
    private static final Command.Action[] STATIC_ACTIONS = {Command.Action.UPDOWN, Command.Action.JUMP, Command.Action.TIP_TOE, Command.Action.JITTER, Command.Action.ASCENDING_TURN, Command.Action.SWING};
    private static final Command.Direction[] STATIC_DIRECTIONS = new Command.Direction[0];

    public StaticCommand(Command.Action action, long duration) {
        setAction(action);
        setDuration(duration);
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command.Action[] getAllowedActions() {
        return STATIC_ACTIONS;
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command.Direction[] getAllowedDirections() {
        return STATIC_DIRECTIONS;
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command copy() {
        return new StaticCommand(getAction(), getDuration().longValue());
    }
}

File: zowi-core/src/main/kotlin/com/bq/zowi/models/commands/StaticCommand.kt
package com.bq.zowi.models.commands

class StaticCommand(action: Command.Action, duration: Long) : MovementCommand() {

    init {
        setAction(action)
        setDuration(duration)
    }

    override fun getAllowedActions(): Array<Command.Action> = STATIC_ACTIONS
    override fun getAllowedDirections(): Array<Command.Direction> = emptyArray()
    override fun copy(): Command = StaticCommand(getAction(), getDuration()!!)
    override fun isRepeatable(): Boolean = false

    companion object {
        private val STATIC_ACTIONS = arrayOf(
            Command.Action.UPDOWN, Command.Action.JUMP, Command.Action.TIP_TOE,
            Command.Action.JITTER, Command.Action.ASCENDING_TURN, Command.Action.SWING
        )
    }
}
